### DISCUSSION TITLE
How to use osm_flutter with local OSM server

### DISCUSSION THREAD

**Discussion by mazenlhm** [liodali/osm_flutter#266] [Q&A] [answered]
Can I use local osm server instead of calling https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png
to our local server such as myserver/osm_tiles/{z}/{x}/{y}.png
and for routing instead of https://router.project-osrm.org/route/v1 I use https://myserver.com/route/v1
As I am using it in a system with strict policy that disallow whitelisting of external ips

**Comment by liodali** [comment, marked as answer]
i putedt already in my list of features
i will try to add it at least for android first as soon as possible

**Comment by oleksandr-lysak** [comment]
Hi, is possible to use my own server for routing at this moment?

**Reply by liodali** [reply]
you can use own client and use manual api to drad road which will change soon 
or you can use our osrm routing client if your server is osrm or valhalla we support both
